docs/proposals/security-context-constraints.md

```diff
@@ -0,0 +1,322 @@
+	RunAsUser RunAsUserStrategyOptions
+
+	// The users who have permissions to use this pod security policy.
+	Users []string
+	// The groups that have permission to use this pod security policy.
+	Groups []string
+}
+
+// SELinuxContextStrategyOptions defines the strategy type and any options used to create the strategy.
+type SELinuxContextStrategyOptions struct {
+	// Type is the strategy that will dictate what SELinux context is used in the SecurityContext.
+	Type SELinuxContextStrategyType
+	// seLinuxOptions required to run as; required for MustRunAs
+	SELinuxOptions *SELinuxOptions
+}
+
+
+// RunAsUserStrategyOptions defines the strategy type and any options used to create the strategy.
+type RunAsUserStrategyOptions struct {
+	// Type is the strategy that will dictate what RunAsUser is used in the SecurityContext.
+	Type RunAsUserStrategyType
+	// UID is the user id that containers must run as.  Required for the MustRunAs strategy if not using
+	// namespace/service account allocated uids.
+	UID *int64
+	// UIDRangeMin defines the min value for a strategy that allocates by range.
+	UIDRangeMin *int64
+	// UIDRangeMax defines the max value for a strategy that allocates by range.
+	UIDRangeMax *int64
+}
+
+// SELinuxContextStrategyType denotes strategy types for generating SELinux options for a
+// SecurityContext
+type SELinuxContextStrategyType string
+
+// RunAsUserStrategyType denotes strategy types for generating RunAsUser values for a
+// SecurityContext
+type RunAsUserStrategyType string
+
+const (
+	// container must have SELinux labels of X applied.
+	SELinuxStrategyMustRunAs SELinuxContextStrategyType = "MustRunAs"
+	// container may make requests for any SELinux context labels.
+	SELinuxStrategyRunAsAny SELinuxContextStrategyType = "RunAsAny"
+
+	// container must run as a particular uid.
+	RunAsUserStrategyMustRunAs RunAsUserStrategyType = "MustRunAs"
+	// container must run as a particular uid.
+	RunAsUserStrategyMustRunAsRange RunAsUserStrategyType = "MustRunAsRange"
+	// container must run as a non-root uid
+	RunAsUserStrategyMustRunAsNonRoot RunAsUserStrategyType = "MustRunAsNonRoot"
+	// container may make requests for any uid.
+	RunAsUserStrategyRunAsAny RunAsUserStrategyType = "RunAsAny"
+)
+
+// PodSecurityPolicyList is a list of PodSecurityPolicy objects
+type PodSecurityPolicyList struct {
+	kapi.TypeMeta
+	kapi.ListMeta
+
+	Items []PodSecurityPolicy
+}
+```
+
+### PodSecurityPolicy Lifecycle
+
+As reusable objects in the root scope, PodSecurityPolicy follows the lifecycle of the
+cluster itself.  Maintenance of constraints such as adding, assigning, or changing them is the
+responsibility of the cluster administrator.
+
+Creating a new user within a namespace should not require the cluster administrator to
+define the user's PodSecurityPolicy.  They should receive the default set of policies
+that the administrator has defined for the groups they are assigned.
```

Comment:
In our current model we assume that that's a function of the authenticator
- if the authenticator returns a set of groups for a user, we can compare
it against this list.  The details of expansion are hidden.

On Mon, Sep 14, 2015 at 2:05 PM, Mike Danese <notifications@github.com>
wrote:

> In docs/proposals/security-context-constraints.md
> <https://github.com/kubernetes/kubernetes/pull/7893#discussion_r39425961>:
>
> > +	kapi.TypeMeta
> > +	kapi.ListMeta
> > +
> > +	Items []PodSecurityPolicy
> > +}
> > +```
> > +
> > +### PodSecurityPolicy Lifecycle
> > +
> > +As reusable objects in the root scope, PodSecurityPolicy follows the lifecycle of the
> > +cluster itself.  Maintenance of constraints such as adding, assigning, or changing them is the
> > +responsibility of the cluster administrator.
> > +
> > +Creating a new user within a namespace should not require the cluster administrator to
> > +define the user's PodSecurityPolicy.  They should receive the default set of policies
> > +that the administrator has defined for the groups they are assigned.
>
> Can a group contain another group as a member?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/7893/files#r39425961>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift